pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -83,6 +83,16 @@ type ServiceMonitorSpec struct {
 
 	// Label selector to select the Kubernetes `Endpoints` objects to scrape metrics from.
 	Selector metav1.LabelSelector `json:"selector"`
+
+	// Mechanism used to select the endpoints to scrape.
+	// By default, the selection process relies on relabel configurations to filter targets.
+	// Alternatively, you can opt for role selectors, which may offer better efficiency in large clusters.
+	// Using role selectors bypasses the need for complex relabeling, making it more suitable for high-scale environments.
+	//
+	// +kubebuilder:validation:Enum=RelabelConfig;RoleSelector
+	// +kubebuilder:default=RelabelConfig
+	SelectorMechanism SelectorMechanism `json:"selectorMechanism,omitempty"`
```

Comment:
still valid?

Reply:
IIRC yes, we need expose a selection option for the users, from what I understood from our call we're going to keep this field and if selected Role it will switch behind the scenes the same role

Reply:
sorry I was dense: I'd advise to have a pointer field and handle the default value in the controller directly (instead of having the `+kubebuilder:default=RelabelConfig` marker).